sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs

```diff
@@ -0,0 +1,435 @@
+        protected const string SignedResourceFullIdentifierToken = AuthenticationTypeToken + " " + SignedResourceToken;
+
+        /// <summary>The character used to separate a token and its value in the connection string.</summary>
+        protected const char TokenValueSeparator = '=';
+
+        /// <summary>The character used to mark the beginning of a new token/value pair in the signature.</summary>
+        protected const char TokenValuePairDelimiter = '&';
+
+        /// <summary>Represents the Unix epoch time value, January 1, 1970 12:00:00, UTC.</summary>
+        protected static readonly DateTime Epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        /// <summary>
+        ///   The name of the shared access key, either for the Event Hubs namespace
+        ///   or the Event Hub.
+        /// </summary>
+        ///
+        public string SharedAccessKeyName { get; protected set; }
+
+        /// <summary>
+        ///   The value of the shared access key, either for the Event Hubs namespace
+        ///   or the Event Hub.
+        /// </summary>
+        ///
+        public string SharedAccessKey { get; protected set;}
+
+        /// <summary>
+        ///   The date and time that the shared access signature expires, in UTC.
+        /// </summary>
+        ///
+        public DateTime ExpirationUtc { get; protected set; }
+
+        /// <summary>
+        ///   The resource to which the shared access signature is intended to serve as
+        ///   authorization.
+        /// </summary>
+        ///
+        public string Resource { get; protected set; }
+
+        /// <summary>
+        ///   The shared access signature to be used for authorization, either for the Event Hubs namespace
+        ///   or the Event Hub.
+        /// </summary>
+        ///
+        public string Value { get; protected set; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="SharedAccessSignature"/> class.
+        /// </summary>
+        ///
+        /// <param name="connectionType">The type of connection that will be used for communicating with the Event Hubs service.</param>
+        /// <param name="host">The fully qualified host name for the Event Hubs namespace.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubPath">The path of the specific Event Hub to connect the client to.</param>
+        /// <param name="sharedAccessKeyName">The name of the shared access key that the signature should be based on.</param>
+        /// <param name="sharedAccessKey">The value of the shared access key for the signagure.</param>
+        /// <param name="signatureValidityDuration">The duration that the signature should be considered valid.</param>
+        ///
+        public SharedAccessSignature(ConnectionType connectionType,
+                                     string         host,
```

Comment:
nit: formatting. This seems to be slightly unconventional formatting for the repo. I'm not sure if we have anything in .editorconfig that covers this but I don't think I've seen other code like this in the repo.

Reply:
I'm a fan of splitting parameters to aid readability.  I've not seen anything that has been consistent guidance around line lengths or always/never breaking things up.   I'm happy to revisit should we standardize on something, but within scope of the preview timeline, I'm going to keep as-is.

Reply:
Reformatted to remove alignment.   I have a feeling that this is another rabbit hole where I'd find myself one of the lone dissenters to other opinions, but I'll mention conversationally (with no ill intent) that there are several studies that support alignment as allowing offering better readability, especially for those more novice to a language.

Reply:
I'm completely open to discussing the coding guideline, but I'm with you on that we need to do it outside of tight deadlines. This is why I'm trying to avoid introducing new concepts in these PRs.